src/lib/provable/int.ts

```diff
@@ -1048,10 +1048,25 @@ class Int64 extends CircuitValue implements BalanceChange {
   // The second point is one of the main things an Int64 is used for, and was the original motivation to use 2 fields.
   // Overall, I think the existing implementation is the optimal one.
 
+  /**
+   * Safely creates a new Int64 instance, enforcing canonical representation of zero.
+   * This is the recommended way to create Int64 instances.
+   *
+   * @param magnitude - The magnitude of the integer as a UInt64
+   * @param sgn - The sign of the integer.
+   * @returns A new Int64 instance with a canonical representation.
+   *
+   * @example
+   * ```ts
+   * const x = new Int64(0); // canonical representation of zero
+   * ```
+   */
   constructor(magnitude: UInt64, sgn = Sign.one) {
-    super(magnitude, sgn);
+    const mag = UInt64.from(magnitude);
+    const isZero = mag.equals(UInt64.zero);
+    const canonicalSign = Provable.if(isZero, Sign.one, sgn);
+    super(mag, canonicalSign);
```

Comment:
this shouldn't be in the constructor logic, should be a separate method like `.create()` as before

the Int64 class still needs an unsafe constructor internally which doesn't add any unnecessary constraints

the the constructor should be made `private`